virtcontainers/clh.go

```diff
@@ -0,0 +1,1260 @@
+	clh.cliBuilder.SetRng(&CLIRng{
+		src:   clh.config.EntropySource,
+		iommu: false,
+	})
+
+	// Add the hybrid vsock device to hypervisor
+	clh.cliBuilder.SetVsock(&CLIVsock{
+		cid:        3,
+		socketPath: clh.socketPath,
+		iommu:      false,
+	})
+
+	// set the initial root/boot disk of hypervisor
+	imagePath, err := clh.config.ImageAssetPath()
+	if err != nil {
+		return err
+	}
+
+	if imagePath != "" {
+		clh.cliBuilder.SetDisk(&CLIDisk{
+			path:  imagePath,
+			iommu: false,
+		})
+	}
+
+	// set the virtio-fs to the hypervisor
+	vfsdSockPath, err := clh.vfsdFSSocketPath(clh.id)
+	if err != nil {
+		return err
+	}
+	clh.cliBuilder.SetFs(&CLIFs{
+		tag:        "kataShared",
+		socketPath: vfsdSockPath,
+		queues:     1,
+		queueSize:  512,
+	})
+
+	// set the serial console to the cloud hypervisor
+	if clh.config.Debug {
+		serialPath, err := clh.serialPath(clh.id)
+		if err != nil {
+			return err
+		}
+		clh.cliBuilder.SetSerial(&CLISerialConsole{
+			consoleType: cctFILE,
+			filePath:    serialPath,
+		})
+	}
+	clh.cliBuilder.SetConsole(&CLIConsole{
+		consoleType: cctOFF,
+	})
+
+	return nil
+}
+
+func (clh *cloudHypervisor) startSandbox(timeout int) error {
+	span, _ := clh.trace("startSandbox")
+	defer span.Finish()
+
+	clh.Logger().WithField("function", "startSandbox").Info("starting Sandbox")
+
+	vmPath := filepath.Join(store.RunVMStoragePath(), clh.id)
+	err := os.MkdirAll(vmPath, store.DirMode)
+	if err != nil {
+		return err
+	}
+
+	if clh.config.SharedFS == config.VirtioFS {
+		clh.Logger().WithField("function", "startSandbox").Info("Starting virtiofsd")
+		timeout, err = clh.setupVirtiofsd(timeout)
+		if err != nil {
+			return err
+		}
+		if err = clh.storeState(); err != nil {
+			return err
+		}
+	}
+
+	var strErr string
+	var pid = 0
+	strErr, err, pid = clh.LaunchIch()
+	if err != nil {
+		return fmt.Errorf("failed to launch cloud-hypervisor: %s, error messages from log: %s", err, strErr)
+	}
+	if err := clh.waitVMM(clhTimeout); err != nil {
+		clh.Logger().WithField("error", err).Warn("cloud-hypervisor init failed")
+		clh.shutdownVirtiofsd()
+		return err
+	}
+
+	clh.state.PID = pid
+	clh.state.state = clhReady
+	clh.storeState()
+
+	return nil
+}
+
+func (clh *cloudHypervisor) getSandboxConsole(id string) (string, error) {
+	clh.Logger().WithField("function", "getSandboxConsole").WithField("id", id).Info("Get Sandbox Console")
+	return "", nil
+}
+
+func (clh *cloudHypervisor) disconnect() {
+	clh.Logger().WithField("function", "disconnect").Info("Disconnecting Sandbox Console")
+}
+
+func (clh *cloudHypervisor) getThreadIDs() (vcpuThreadIDs, error) {
+
+	clh.Logger().WithField("function", "getThreadIDs").Info("get thread ID's")
+
+	var vcpuInfo vcpuThreadIDs
+
+	vcpuInfo.vcpus = make(map[int]int)
+
+	return vcpuInfo, nil
+}
+
+func (clh *cloudHypervisor) hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	clh.Logger().WithField("function", "hotplugAddDevice").Info("Add hotplug device")
```

Comment:
The message should indicate that it's not supported. And I would suggest to use a `Warn` instead of `Info`.